src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,46 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_ARRAY_SIZE = 10;
+    private final K[] keys;
+    private final V[] values;
+    private int size;
+
+    public StorageImpl() {
+        keys = (K[]) new Object[MAX_ARRAY_SIZE];
+        values = (V[]) new Object[MAX_ARRAY_SIZE];
+        size = 0;
+    }
+
     @Override
     public void put(K key, V value) {
+        for (int i = 0; i <= size(); i++) {
+            if (areKeysEqual(keys[i], key) && values[i] != null) {
+                values[i] = value;
+                return;
+            }
+        }
+        keys[size] = key;
+        values[size] = value;
+        size++;
     }
 
     @Override
     public V get(K key) {
+        for (int j = 0; j < size(); j++) {
+            if (areKeysEqual(keys[j], key)) {
+                return values[j];
+            }
+        }
```

Comment:
Let's extract it to a separate method.

```suggestion
private int indexOf(key) {
        for (int j = 0; j < size(); j++) {
            if (areKeysEqual(keys[j], key)) {
                return i;
            }
        }
}        
```